apps/system/js/browser.js

```diff
@@ -1,9 +1,17 @@
-/* global UrlHelper, AppWindow, BrowserConfigHelper, LazyLoader */
+/* global UrlHelper, AppWindow, BrowserConfigHelper, LazyLoader,
+          SettingsListener */
 
 (function(exports) {
 
   'use strict';
 
+  function handleSettingChange (value) {
+    Browser.privateByDefault = value;
+  }
+
+  SettingsListener.observe('browser.private.default', false,
```

Comment:
The listeners in this file are currently in the start method of the Browser "class". To keep things in line with the rest of the system app, I might recommend moving this listener to into the start() method as well - it should make it easier to write a unit test against this if we wanted to in the future. (If this goes into the start method, then privateByDefault would probably go onto Browser.prototype).